src/DataSourcePoller.php

```diff
@@ -0,0 +1,231 @@
+<?php
+
+namespace Automattic\WooCommerce\Admin;
+
+/**
+ * Specs data source poller class.
+ * This handles polling specs from JSON endpoints, and
+ * stores the specs in to the database as an option.
+ */
+class DataSourcePoller {
```

Comment:
I see we're using this class in a couple different ways.  In one case, we use this as a generic class with arguments while another we use it to extend a child class.

I think it might make it easier to follow by deciding on one of these and sticking to a convention.  I think both routes are solid and comes with perks.